compiler/hash-typecheck/src/ops/pats.rs

```diff
@@ -183,8 +183,18 @@ impl<'tc> PatMatcher<'tc> {
                     Err(_) => Ok(None),
                 }
             }
+            Pat::Range { lo, hi, .. } => {
+                // The `lo` and `hi` should always unify with the subject term...
+                self.unifier().unify_terms(lo, simplified_term_id)?;
+                self.unifier().unify_terms(hi, simplified_term_id)?;
+
+                // They should also unify with one another...
+                self.unifier().unifier().unify_terms(hi, lo)?;
```

Comment:
This should happen in pattern validation (a function on validator, or infer ty of pat ?).

Can you also add a @@Fixme at the top of this function to rework level 0 unification (i.e. should never work when Rts are involved), and that this function should unify the *types*, rather than the values, of the pattern and term.